app/api/dao/user.py

```diff
@@ -418,3 +421,80 @@ def get_user_statistics(user_id):
             'achievements': achievements
         }
         return response
+
+    @staticmethod
+    def get_user_dashboard(user_id):
+        user = UserModel.find_by_id(user_id)
+
+        if not user:
+            return None
+
+        current_relation = MentorshipRelationDAO.list_current_mentorship_relation(user_id)
+        all_relations = user.mentor_relations + user.mentee_relations
+
+        relations_received_as_mentee = [rel for rel in all_relations if rel.action_user_id != user_id
+                                        and rel.mentee_id == user_id]
+        relations_sent_as_mentee = [rel for rel in all_relations if rel.action_user_id == user_id
+                                    and rel.mentee_id == user_id]
+        relations_received_as_mentor = [rel for rel in all_relations if rel.action_user_id != user_id
+                                        and rel.mentor_id == user_id]
+        relations_sent_as_mentor = [rel for rel in all_relations if rel.action_user_id == user_id
+                                    and rel.mentor_id == user_id]
+
+        current_tasks = []
+        if isinstance(current_relation, MentorshipRelationModel):
+            current_tasks = TaskDAO.list_tasks(user_id, current_relation.id)
+
+        dashboard = {
+            # relations received as mentee, by state
+            "pending_relations_received_as_mentee": [rel.json() for rel in relations_received_as_mentee if
```

Comment:
just wondering if this can be done bit more lazy, by looping all states and generating keyname of lowercase of enum `+ _relations_received_as_mentee` ?

Reply:
Well, it would be possible, but I already did it this way and works fine.
It's more explicit now and...I think this part of code is complex enough, I'd prefer not to throw even more there.

Reply:
Fair enough.

Reply:
@annabauza I have improved this code a bit, you can check out the newest commit.